platformio.ini

```diff
@@ -37,5 +37,5 @@ platform = espressif8266
 board = esp12e
 framework = arduino
 lib_deps = ${common.lib_deps}
-src_build_flags = ${common.version} -DENABLE_OTA -DWIFI_LED=0 -DENABLE_DEBUG
-upload_port = openevse.local
+src_build_flags = ${common.version} -DENABLE_OTA -DWIFI_LED=0 -DENABLE_DEBUG -DHOSTNAME=openevse-ota
```

Comment:
Not totally sure I like this, really what we need is some way to set this locally outside of what is checked in, or ideally, as you say save it in the settings.

You can use ${env.XXXX} to pull in system [environment vars](http://docs.platformio.org/en/latest/projectconf.html#dynamic-variables) but I don't see a way of setting a default when not set. Another option may be to use a [script ](http://docs.platformio.org/en/latest/projectconf.html#dynamic-variables) to read in the host name from a file, giving a default if not there?